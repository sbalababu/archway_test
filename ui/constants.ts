export const TOKEN_EXTRACTOR = (s: any) => s.getIn(['login', 'token']);

export const RECENT_WORKSPACES_KEY = 'recentWorkspaces';
export const SPNEGO = 'spnego';
export const workspaceStatuses = ['approved', 'pending', 'rejected'];
export const workspaceBehaviors = ['simple', 'structured', 'custom'];
export const behaviorProperties = {
  simple: {
    title: 'Simple',
    icon: 'team',
  },
  structured: {
    title: 'Structured',
    icon: 'deployment-unit',
  },
  custom: {
    title: 'Custom',
    icon: 'select',
  },
};

export enum FeatureFlagType {
  Application = 'application',
  Messaging = 'messaging',
  DiskQuota = 'disk-quota',
}

export enum ProvisioningType {
  Pending = 'pending',
  Complete = 'complete',
}

export enum ModalType {
  DeprovisionWorkspace = 'deprovisionWorkspace',
  ProvisionWorkspace = 'provisionWorkspace',
  DeleteWorkspace = 'deleteWorkspace',
  Application = 'application',
  Kafka = 'kafka',
  SimpleTopicMember = 'simpleTopicMember',
  SimpleMember = 'simpleMember',
  ChangeOwner = 'changeOwner',
  ModifyDiskQuota = 'modifyDiskQuota',
  ModifyCoreMemory = 'modifyCoreMemory',
}

export enum NotificationType {
  Success = 'success',
  Info = 'info',
  Warning = 'warning',
  Error = 'error',
}

export enum ComplianceType {
  PCI = 'pci',
  PII = 'pii',
  PHI = 'phi',
}

export enum LinksGroupCardPage {
  Overview = 'home',
  Manage = 'manage',
}
