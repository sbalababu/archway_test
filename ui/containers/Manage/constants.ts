export enum ManagePage {
  ComplianceTab = 'compliances',
  LinksTab = 'custom-links',
}

export enum OpenType {
  Add = 'add',
  Update = 'update',
  Delete = 'delete',
}

export const formLayout = {
  labelCol: {
    xs: { span: 24 },
    sm: { span: 4 },
  },
  wrapperCol: {
    xs: { span: 24 },
    sm: { span: 20 },
    lg: { span: 16 },
  },
};

export const linkLayout = {
  labelCol: {
    xs: { span: 24 },
    sm: { span: 3 },
  },
  wrapperCol: {
    xs: { span: 24 },
    md: { span: 21 },
    lg: { span: 15 },
  },
};
