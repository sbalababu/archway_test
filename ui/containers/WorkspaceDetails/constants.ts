export enum ShowTypes {
  Users = 'Users',
  Groups = 'Groups',
}
