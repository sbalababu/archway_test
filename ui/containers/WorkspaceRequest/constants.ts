export const PAGE_BEHAVIOR = 'behavior';
export const PAGE_CUSTOM_WORKSPACES = 'custom workspaces';
export const PAGE_DETAILS = 'details';
export const PAGE_COMPLIANCE = 'compliance';
export const PAGE_REVIEW = 'review';
