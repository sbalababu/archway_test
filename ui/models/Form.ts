export interface QuotaForm {
  quota: number;
}

export interface CoreMemoryForm {
  core: number;
  memory: number;
}

export interface MemberForm {
  username: string;
  roles: object;
}
