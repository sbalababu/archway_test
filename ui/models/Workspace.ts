export interface YarnApplication {
  id: string;
  name: string;
  start_time?: Date;
}

export interface ResourcePoolsInfo {
  loading: boolean;
  error?: string;
  data: PoolInfo[];
}

export interface PoolInfo {
  resource_pool: string;
  applications: YarnApplication[];
}

export interface NamespaceInfo {
  name: string;
  tables: HiveTable[];
}

export interface NamespaceInfoList {
  loading: boolean;
  error?: string;
  data: NamespaceInfo[];
}

export interface HiveTable {
  name: string;
}

export interface MemberRemoveStatus {
  loading?: boolean;
  success?: boolean;
  error?: boolean;
}

export interface Member {
  distinguished_name: string;
  email?: string;
  name: string;
  removeStatus?: MemberRemoveStatus;
  data: any;
  topics: any;
  applications: any;
}

export interface Application {
  id: number;
  name: string;
  consumer_group: string;
  group: SecurityGroup;
  type?: string;
  logo?: string;
  language?: string;
  repository?: string;
}

export interface TopicGrant {
  id: number;
  group: SecurityGroup;
  actions: string;
  topic_access?: Date;
}

export interface KafkaTopic {
  id: number;
  name: string;
  partitions: number;
  replication_factor: number;
  managing_role: TopicGrant;
  readonly_role: TopicGrant;
}

export interface ResourcePool {
  id: number;
  pool_name: string;
  max_cores: number;
  max_memory_in_gb: number;
}

export interface SecurityGroup {
  common_name: string;
  distinguished_name: string;
  sentry_role: string;
  group_created?: Date;
  role_created?: Date;
  role_associated?: Date;
  attributes?: string[][];
}

export interface Compliance {
  phi_data?: boolean;
  pci_data?: boolean;
  pii_data?: boolean;
}

export interface DatabaseGrant {
  group: SecurityGroup;
  location_access?: Date;
  database_access?: Date;
}

export interface HiveAllocation {
  id: number;
  name: string;
  location: string;
  size_in_gb: number;
  consumed_in_gb: number;
  managing_group: DatabaseGrant;
  readwrite_group?: DatabaseGrant;
  readonly_group?: DatabaseGrant;
}

export interface ApprovalStatus {
  loading?: boolean;
  success?: boolean;
  error?: string;
}

export interface ApprovalItem {
  approver?: string;
  approval_time?: Date;
  status?: ApprovalStatus;
}

export interface Approvals {
  infra?: ApprovalItem;
  risk?: ApprovalItem;
}

export interface Workspace {
  id: number;
  name: string;
  summary: string;
  description: string;
  behavior: string;
  requested_date: Date;
  requester: string;
  single_user: boolean;
  status?: string;
  compliance: Compliance;
  approvals?: Approvals;
  data: HiveAllocation[];
  processing: ResourcePool[];
  topics: KafkaTopic[];
  applications: Application[];
}

export interface UserSuggestion {
  display: string;
  distinguished_name: string;
}

export interface UserSuggestions {
  filter: string;
  users: UserSuggestion[];
  groups: UserSuggestion[];
}

export interface WorkspaceSearchResult {
  id: number;
  name: string;
  summary: string;
  status: string;
  behavior: string;
  pii_data?: boolean;
  pci_data?: boolean;
  phi_data?: boolean;
  date_requested: Date;
  date_fully_approved: Date;
  total_disk_allocated_in_gb: number;
  total_disk_consumed_in_gb?: number;
  total_max_cores: number;
  total_max_memory_in_gb: number;
}
