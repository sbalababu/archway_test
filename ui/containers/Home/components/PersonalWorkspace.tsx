import * as React from 'react';
import { Button, Card, Spin, Row, Col } from 'antd';
import { PrepareHelp, RunHelp, CreateHelp, Label } from './';
import { Provisioning } from '../../../components';
import { Workspace } from '../../../models/Workspace';
import { Cluster } from '../../../models/Cluster';
import { ProvisioningType, FeatureFlagType } from '../../../constants';
import { Feature } from '../../../components';

interface Props {
  workspace: Workspace;
  services: any;
  loading: boolean;
  cluster?: Cluster;
  provisioning: ProvisioningType;
  workspaceFetched: boolean;
  requestWorkspace: () => void;
}

const PersonalWorkspace = ({
  workspace,
  services,
  requestWorkspace,
  loading,
  provisioning,
  workspaceFetched,
}: Props) => {
  if (loading) {
    return (
      <div
        style={{
          display: 'flex',
          justifyContent: 'center',
          alignItems: 'center',
          padding: 24,
        }}
      >
        <Spin />
      </div>
    );
  }

  return (
    <div style={{ flex: 1 }}>
      {workspace && workspace.data ? (
        <Card bodyStyle={{ display: 'flex', flexDirection: 'column' }}>
          <Label>Your Personal Workspace</Label>
          <Provisioning provisioning={provisioning} />
          <div>
            <div
              style={{
                display: 'flex',
                justifyContent: 'center',
                textAlign: 'center',
                fontWeight: 'bold',
                fontSize: '12px',
              }}
            >
              <div style={{ padding: '10px' }}>
                <div>HDFS LOCATION</div>
                <div style={{ fontWeight: 200 }}>{workspace.data[0].location}</div>
              </div>
              <div style={{ padding: '10px' }}>
                <div>HIVE NAMESPACE</div>
                <div style={{ fontWeight: 200 }}>{workspace.data[0].name}</div>
              </div>
              <Feature flag={FeatureFlagType.Application}>
                <div style={{ padding: '10px' }}>
                  <div>RESOURCE POOL</div>
                  <div style={{ fontWeight: 200 }}>{workspace.processing[0].pool_name}</div>
                </div>
              </Feature>
              {!!workspace.topics[0] && !!workspace.topics[0].name && (
                <Feature flag={FeatureFlagType.Messaging}>
                  <div style={{ padding: '10px' }}>
                    <div>KAFKA TOPIC</div>
                    <div style={{ fontWeight: 200 }}>{workspace.topics[0].name}</div>
                  </div>
                </Feature>
              )}
            </div>
            <Row gutter={12}>
              <Col span={24} xl={8} style={{ marginTop: 10 }}>
                <PrepareHelp location={workspace.data[0].location} namespace={workspace.data[0].name} />
              </Col>
              <Col span={24} xl={8} style={{ marginTop: 10 }}>
                <CreateHelp
                  host={services.hive.thrift && services.hive.thrift[0].host}
                  port={services.hive.thrift && services.hive.thrift[0].port}
                  namespace={workspace.data[0].name}
                />
              </Col>
              <Col span={24} xl={8} style={{ marginTop: 10 }}>
                <RunHelp queue={workspace.processing[0].pool_name} />
              </Col>
            </Row>
          </div>
        </Card>
      ) : (
        <Card bodyStyle={{ textAlign: 'center' }}>
          {workspaceFetched ? (
            <React.Fragment>
              <h3>You don't have a personal workspace yet!</h3>
              <Button type="primary" onClick={requestWorkspace}>
                Create One Now
              </Button>
            </React.Fragment>
          ) : (
            <h3 style={{ color: 'red' }}>Error retrieving personal workspace.</h3>
          )}
        </Card>
      )}
    </div>
  );
};

export default PersonalWorkspace;
