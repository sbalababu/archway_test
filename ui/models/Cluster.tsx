import { Dropdown, Icon, Menu } from 'antd';
import * as Color from 'color';
import * as React from 'react';
import { Colors } from '../components';

export interface WebLocation {
  host: string;
  port: number;
}

export interface Statusable {
  status: string;
}

export class Status<T extends Statusable> {
  public statusable: T;

  constructor(statusable: T) {
    this.statusable = statusable;
  }

  public statusColor = (): Color => {
    switch (this.statusable.status) {
      case 'GOOD_HEALTH':
        return Colors.Green;
      case 'CONCERNING_HEALTH':
        return Colors.Orange;
      case 'BAD_HEALTH':
        return Colors.Red;
      default:
        return Colors.Gray;
    }
  };

  public statusText = () => {
    switch (this.statusable.status) {
      case 'GOOD_HEALTH':
        return 'good';
      case 'CONCERNING_HEALTH':
        return 'concerning';
      case 'BAD_HEALTH':
        return 'bad';
      default:
        return 'unknown';
    }
  };

  public glowColorText = () =>
    `0 0 5px 2px ${this.statusColor()
      .hsl()
      .string()}`;
}

export interface HueService extends Statusable {
  load_balancer: WebLocation[];
}

/* tslint:disable:no-empty-interface */
export interface HiveService extends Statusable {
  thrift: WebLocation[];
}

export interface YarnService extends Statusable {
  resource_manager: WebLocation[];
  node_manager: WebLocation[];
}

export interface NavigatorService extends Statusable {
  navigator: WebLocation[];
}

export abstract class ServiceLinks<T extends Statusable> {
  public service: T;

  public links!: JSX.Element[];

  constructor(service: T) {
    this.service = service;
  }
}

export class HueServiceLinks extends ServiceLinks<HueService> {
  public hueServiceLinks: JSX.Element[] =
    this.service &&
    this.service.load_balancer.map(location => (
      <Menu.Item key={location.host}>
        <a target="_blank" rel="noreferrer noopener" href={`https://${location.host}:${location.port}`}>
          Hue UI
        </a>
      </Menu.Item>
    ));

  public links: JSX.Element[] = [
    // tslint:disable-next-line: jsx-key
    <Dropdown overlay={<Menu>{this.hueServiceLinks}</Menu>}>
      <a href="#" className="ant-dropdown-link">
        {/* eslint-disable-line */}
        Quick Links <Icon type="down" />
      </a>
    </Dropdown>,
  ];
}

export class HiveServiceLinks extends ServiceLinks<HiveService> {
  public links: JSX.Element[] = [0].map(index => (
    <h4 key={index} style={{ cursor: 'not-allowed', color: 'rgb(0,0,0,.45)', margin: 0 }}>
      (no links)
    </h4>
  ));
}

export class YarnServiceLinks extends ServiceLinks<YarnService> {
  public resourceManagerLinks: JSX.Element[] =
    this.service &&
    this.service.resource_manager.map(location => (
      <Menu.Item key={location.host}>
        <a target="_blank" rel="noreferrer noopener" href={`https://${location.host}:${location.port}`}>
          Resource Manager UI ({location.host})
        </a>
      </Menu.Item>
    ));

  public nodeManagerLinks: JSX.Element[] =
    this.service &&
    this.service.node_manager.map(location => (
      <Menu.Item key={location.host}>
        <a target="_blank" rel="noreferrer noopener" href={`https://${location.host}:${location.port}`}>
          Node Manager UI ({location.host})
        </a>
      </Menu.Item>
    ));

  public links: JSX.Element[] = [
    // tslint:disable-next-line: jsx-key
    <Dropdown
      overlay={
        <Menu>
          {this.resourceManagerLinks}
          {this.nodeManagerLinks}
        </Menu>
      }
    >
      <a href="#" className="ant-dropdown-link">
        {/* eslint-disable-line */}
        Quick Links <Icon type="down" />
      </a>
    </Dropdown>,
  ];
}

export class NavigatorServiceLinks extends ServiceLinks<NavigatorService> {
  public navigatorServiceLinks: JSX.Element[] =
    this.service &&
    this.service.navigator.map(location => (
      <Menu.Item key={location.host}>
        <a target="_blank" rel="noreferrer noopener" href={`https://${location.host}:${location.port}`}>
          Navigator UI
        </a>
      </Menu.Item>
    ));

  public links: JSX.Element[] = [
    // tslint:disable-next-line: jsx-key
    <Dropdown overlay={<Menu>{this.navigatorServiceLinks}</Menu>}>
      <a href="#" className="ant-dropdown-link">
        {/* eslint-disable-line */}
        Quick Links <Icon type="down" />
      </a>
    </Dropdown>,
  ];
}

export interface Cluster extends Statusable {
  name: string;
  cm_url: string;
  services: {
    hive: HiveService;
    hue: HueService;
    yarn: YarnService;
    mgmt: NavigatorService;
  };
}
