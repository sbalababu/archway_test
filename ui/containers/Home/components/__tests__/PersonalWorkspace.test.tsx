import * as React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import PersonalWorkspace from '../PersonalWorkspace';
import { Workspace, KafkaTopic } from '../../../../models/Workspace';
import { ProvisioningType } from '../../../../constants';

describe('PersonalWorkspace', () => {
  it('renders correctly', () => {
    const workspace: Workspace = {
      id: 0,
      name: '',
      summary: '',
      description: '',
      behavior: '',
      requested_date: new Date(0),
      requester: '',
      single_user: false,
      compliance: {
        phi_data: false,
        pci_data: false,
        pii_data: false,
      },
      data: [
        {
          id: 1,
          name: '',
          location: '',
          size_in_gb: 0,
          consumed_in_gb: 0,
          managing_group: {
            group: {
              common_name: '',
              distinguished_name: '',
              sentry_role: '',
              attributes: [['', '']],
            },
          },
        },
      ],
      processing: [
        {
          id: 0,
          pool_name: '',
          max_cores: 0,
          max_memory_in_gb: 0,
        },
      ],
      topics: [
        {
          id: 1,
          name: '',
          partitions: 1024,
          replication_factor: 1,
        } as KafkaTopic,
      ],
      applications: [],
    };
    const services = {
      yarn: {
        //
      },
      hue: {
        //
      },
      hive: {
        thrift: [{ host: '', port: '' }],
      },
      impala: {
        //
      },
    };
    const wrapper = shallow(
      <PersonalWorkspace
        workspace={workspace}
        services={services}
        loading={false}
        requestWorkspace={() => null}
        provisioning={ProvisioningType.Complete}
        workspaceFetched={true}
      />
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
