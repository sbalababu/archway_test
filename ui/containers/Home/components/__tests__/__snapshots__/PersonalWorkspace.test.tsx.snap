// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PersonalWorkspace renders correctly 1`] = `
<div
  style={
    Object {
      "flex": 1,
    }
  }
>
  <Card
    bodyStyle={
      Object {
        "display": "flex",
        "flexDirection": "column",
      }
    }
  >
    <Label>
      Your Personal Workspace
    </Label>
    <Provisioning
      provisioning="complete"
    />
    <div>
      <div
        style={
          Object {
            "display": "flex",
            "fontSize": "12px",
            "fontWeight": "bold",
            "justifyContent": "center",
            "textAlign": "center",
          }
        }
      >
        <div
          style={
            Object {
              "padding": "10px",
            }
          }
        >
          <div>
            HDFS LOCATION
          </div>
          <div
            style={
              Object {
                "fontWeight": 200,
              }
            }
          />
        </div>
        <div
          style={
            Object {
              "padding": "10px",
            }
          }
        >
          <div>
            HIVE NAMESPACE
          </div>
          <div
            style={
              Object {
                "fontWeight": 200,
              }
            }
          />
        </div>
        <Connect(Feature)
          flag="application"
        >
          <div
            style={
              Object {
                "padding": "10px",
              }
            }
          >
            <div>
              RESOURCE POOL
            </div>
            <div
              style={
                Object {
                  "fontWeight": 200,
                }
              }
            />
          </div>
        </Connect(Feature)>
      </div>
      <Row
        gutter={12}
      >
        <Col
          span={24}
          style={
            Object {
              "marginTop": 10,
            }
          }
          xl={8}
        >
          <SetupHelp
            location=""
            namespace=""
          />
        </Col>
        <Col
          span={24}
          style={
            Object {
              "marginTop": 10,
            }
          }
          xl={8}
        >
          <SetupHelp
            host=""
            namespace=""
            port=""
          />
        </Col>
        <Col
          span={24}
          style={
            Object {
              "marginTop": 10,
            }
          }
          xl={8}
        >
          <RunHelp
            queue=""
          />
        </Col>
      </Row>
    </div>
  </Card>
</div>
`;
